### ISSUE TITLE
Warm-started batched differentiable solve: batched_solve has no warm seam (follow-up to #74/#76)

### ISSUE THREAD

**Issue by robertojjx** [jkitchin/pounce#78]
## Summary

Follow-up to #74 (closed) and #76. `pounce.jax` now has two warm-start surfaces, but the **batched** differentiable solve — which #76 makes the recommended fast path — is **not** one of them. Request: a batched warm-start entry point, e.g. `JaxProblem.batched_solve_with_warm(p_batch, x0_batch, warm_start_batch)`, that threads per-block primal+dual state `(lam, zL, zU)` into the single stacked solve and recovers it on return.

This is a feature request, not a bug — the current behavior is by design.

## What exists today

Reading `python/pounce/jax/_problem.py` (editable install at `6be3643`-era source):

- `solve_with_warm(p, x0, warm_start=(lam, zL, zU)) -> (x*, (lam, zL, zU))` — differentiable, threads full primal+dual state. **Single-sample only.** It's also the only path that sets `warm_start_init_point="yes"` (via `_thread_problem_warm`).
- `batched_solve(p_batch, x0)` — the stacked block-diagonal solve from #76 (one Rust crossing, one shared factorization, factor-reuse backward). Takes a **primal `x0` only**; `_host_batched_solve` builds a fresh `Solver` and calls `solver.solve(x0=X0)` cold every time. It *returns* the converged `mult_g / mult_x_L / mult_x_U` for the backward pass but never feeds them back in, and never enables `warm_start_init_point`.

So to get both batched speed **and** warm starts today, a caller has to loop `solve_with_warm` per sample — which gives up exactly the single-crossing / shared-factorization advantage `batched_solve` was added for.

## Use case

A differentiable constrained-projection layer inside a NN training loop projects a whole minibatch every step, with the objective `||x - y_hat||²`. The parameter `p` (carrying `y_hat`) drifts only slightly per step, so last step's primal+dual solution is an excellent warm start. Our cyipopt-based backend gets a large per-step speedup precisely by reusing one persistent stacked NLP and calling `nlp.solve(x0, lagrange=lam_prev, zl=zL_prev, zu=zU_prev)` with `warm_start_bound_push=1e-9` — one set of warm vectors covers the whole stacked batch in a single solve. The pounce `batched_solve` path has no equivalent, so every training step pays a cold interior-point solve.

## Requested API

```python
x_batch, (lam_b, zL_b, zU_b) = jp.batched_solve_with_warm(
    p_batch, x0_batch,
    warm_start=(lam_batch, zL_batch, zU_batch),  # or None for a cold first call
)
```

- Per-block warm vectors `(B, m)`, `(B, n)`, `(B, n)` mapped onto the stacked variable/constraint ordering, and `warm_start_init_point="yes"` set on the stacked `Solver` (mirroring `_thread_problem_warm`).
- Returns the converged stacked duals reshaped per-block, so the caller can thread them into the next step — the batched analogue of what `solve_with_warm` already returns single-sample.
- Differentiable like `batched_solve` (warm state need not participate in the gradient — `x0`/duals can be treated as stop-gradient constants, same as `solve_with_warm`'s warm args).

## Why batched specifically (not "just loop solve_with_warm")

The whole point of #76's `batched_solve` is one Rust crossing and one shared symbolic factorization across the block-diagonal KKT, with the factor-reuse backward. Looping `solve_with_warm` reintroduces B crossings and B factorizations, which is the cost #76 was designed to remove. Warm-starting only helps the inner-loop use case if it composes with the batched path rather than forcing a fallback to the per-sample one.

## Acknowledgement

#74 delivered warm start on the single-sample `solve_with_warm` and batching via `batched_solve`/`vmap_solve_parallel` — both very useful. This is just the missing intersection: warm start *on* the batched solve. Filing as an enhancement with the use case spelled out so priority can be judged against it.

**Comment by jkitchin** [comment]
Closed. `JaxProblem.batched_solve_with_warm` is the intersection of `batched_solve` (#76) and `solve_with_warm` (#74): one stacked Rust crossing per batch, with per-block warm `(lambda, z_L, z_U)` threaded in and out.

### API

```python
x_batch, (lam_b, zL_b, zU_b) = jp.batched_solve_with_warm(
    p_batch, x0, warm_start=(lam_prev, zL_prev, zU_prev),
)
```

Pass `warm_start=None` (or omit) on the first call when no prior state is available — same shape as `solve_with_warm`'s single-sample interface.

### Internals

- A new `_thread_stacked_problem_warm` TLS cache holds a dedicated stacked `Problem` with `warm_start_init_point="yes"` so a training loop that interleaves warm and cold batched solves doesn't flip the option mid-life on a shared cached `Problem`.
- Backward is wired with `jax.custom_vjp`; warm vectors and `x0` are stop-gradient (same convention `solve_with_warm` uses); the cotangent solve goes through the held stacked LDLᵀ factor (`factor_reuse=True`) via `kkt_solve_many` — same path #77 just made batched, so the backward picks up that win for free.

### Coverage

- `python/tests/test_jax.py::test_batched_solve_with_warm_matches_loop_pounce_78` — batched warm result matches a per-block `solve_with_warm` loop.
- `python/tests/test_jax.py::test_batched_solve_with_warm_grad_matches_batched_solve_pounce_78` — `jax.grad` through warm matches `jax.grad` through plain `batched_solve` to IPM tolerance.
- `python/tests/test_jax.py::test_batched_solve_with_warm_warm_inputs_are_stop_gradient_pounce_78` — `lam/zL/zU` and `x0` carry no gradient signal.

### Demonstration

Executed notebook `python/notebooks/11_batched_warm_start.ipynb`:
- Simulates a 6-step training loop (per-step parameter drift = 0.05 std-dev Gaussian) on a constrained-projection block.
- **1.20× fewer IPM iters cumulative** when warm-starting (5 iters/step warm vs 6 iters/step cold), on top of the per-step amortisation `batched_solve` already gives.
- `max |grad_warm − grad_cold|` = 4.44e-16 — gradient through the new path matches the gradient through plain `batched_solve` end-to-end.

Closing.
